catalog/spatial/geocoding/spatial-geocoding-geocoder/src/main/java/org/codice/ddf/spatial/geocoder/GeoCoder.java

```diff
@@ -32,4 +34,14 @@
      * @throws GeoEntryQueryException if an exception occurs while querying the GeoNames resource
      */
     NearbyLocation getNearbyCity(String locationWKT) throws GeoEntryQueryException;
+
+    /**
+     * Retrieves the country code for a given location. The center point is used if {@code locationWKT}
+     * is a polygon.
+     *
+     * @param locationWKT WKT location for which to get the country code of
+     * @param radius      Radius in kilometers to search from the center of {@code locationWKT}
+     * @return a country code in ISO 3166-1 alpha-3 format or null if not found (for example, a location in the ocean)
+     */
+    Optional<String> getCountryCode(String locationWKT, int radius);
```

Comment:
Decided we are going to leave this as is and instead implement an Attribute validator for the `Location.COUNTRY_CODE` attribute. This should be done in a separate ticket, and as of right now if a alpha-3 country code is not returned from the `Locale`, the `Location.COUNTRY_CODE` attribute will not be populated, even if we have a valid alpha2 country code, in order to preserve the constraint on this attribute. So this should be safe for now.